Sources/SpeziLLMLocalDownload/LLMLocalDownloadManager.swift

```diff
@@ -79,49 +79,47 @@ public final class LLMLocalDownloadManager: NSObject {
     }
     
     /// Starts a `URLSessionDownloadTask` to download the specified model.
+    @MainActor
     public func startDownload() async {
         if modelExist {
-            Task { @MainActor in
-                self.state = .downloaded
-            }
+            state = .downloaded
             return
         }
         
         await cancelDownload()
         downloadTask = Task(priority: .userInitiated) {
             do {
                 try await downloadWithHub()
-                await MainActor.run {
-                    self.state = .downloaded
-                }
+                state = .downloaded
             } catch {
-                await MainActor.run {
-                    self.state = .error(
-                        AnyLocalizedError(
-                            error: error,
-                            defaultErrorDescription: LocalizedStringResource("LLM_DOWNLOAD_FAILED_ERROR", bundle: .atURL(from: .module))
+                state = .error(
+                    AnyLocalizedError(
+                        error: error,
+                        defaultErrorDescription: LocalizedStringResource(
+                            "LLM_DOWNLOAD_FAILED_ERROR",
+                            bundle: .atURL(from: .module)
                         )
                     )
-                }
+                )
             }
         }
     }
     
     /// Cancels the download of a specified model via a `URLSessionDownloadTask`.
+    @MainActor
     public func cancelDownload() async {
         downloadTask?.cancel()
-        await MainActor.run {
-            self.state = .idle
-        }
+        state = .idle
     }
-    
-    @MainActor
+
     private func downloadWithHub() async throws {
         let repo = Hub.Repo(id: model.hubID)
         let modelFiles = ["*.safetensors", "config.json"]
         
         try await HubApi.shared.snapshot(from: repo, matching: modelFiles) { progress in
```

Comment:
This is forcing me to make the whole class `@unchecked Sendable`, because we can't pass `progress` safely without potentially causing data races. Do you have another idea?